### PR TITLE
Use Stopwatch instead of wall clock for async job timeouts

### PR DESCRIPTION
**Pull request by xPaw**
Seems like a better thing to do, should also hopefully improve randomness in tests.

### CODE REVIEW

File: SteamKit2/SteamKit2/Util/ValueStopwatch.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        public TimeSpan GetElapsedTime()
+        {
+            if ( _startTimestamp == 0 )
+            {
+                throw new InvalidOperationException( "An uninitialized, or 'default', ValueStopwatch cannot be used to get elapsed time." );
+            }
+
+            long end = Stopwatch.GetTimestamp();
+            long timestampDelta = end - _startTimestamp;
+            long ticks = ( long )( s_timestampToTicks * timestampDelta );
```
Comment (yaakov-h) [inline SteamKit2/SteamKit2/Util/ValueStopwatch.cs:28]: isn't the timestamp supposed to already be in Ticks?

Comment (xPaw) [reply SteamKit2/SteamKit2/Util/ValueStopwatch.cs:28]: This is straight from .NET runtime: https://source.dot.net/#Microsoft.Extensions.Http/ValueStopwatch.cs,510774ca21949635

File: SteamKit2/SteamKit2/Types/JobID.cs [hunk 2]
```diff
@@ -6,10 +6,12 @@
 using System;
 using System.Collections.Generic;
 using System.Collections.ObjectModel;
+using System.Diagnostics;
```
Comment (yaakov-h) [inline SteamKit2/SteamKit2/Types/JobID.cs:9]: [where] is this needed?

Comment (xPaw) [reply SteamKit2/SteamKit2/Types/JobID.cs]: its not. gonna need to do a usings cleanup pass, because a lot of files have unused ones.

**Pull request merged**